typescript/src/actions/solana/transfer.ts

```diff
@@ -0,0 +1,223 @@
+import {
+  Connection,
+  MessageV0,
+  PublicKey,
+  SystemProgram,
+  TransactionInstruction,
+  TransactionMessage,
+  VersionedTransaction,
+} from "@solana/web3.js";
+
+import { getConnectedNetwork, getOrCreateConnection, type Network } from "./utils.js";
+
+import type { SignatureResult } from "../../client/solana/solana.types.js";
+import type { CdpOpenApiClientType } from "../../openapi-client/index.js";
+
+export interface TransferOptions {
+  /**
+   * The base58 encoded Solana address of the source account.
+   */
+  from: string;
+  /**
+   * The base58 encoded Solana address of the destination account.
+   */
+  to: string;
+  /**
+   * The amount to transfer, represented as an atomic unit of the token.
+   */
+  amount: bigint;
+  /**
+   * The token to transfer, or mint address of the SPL token to transfer.
+   */
+  token: "sol" | "usdc" | string;
+  /**
+   * The network to use which will be used to create a Connection, otherwise a Connection can be provided.
+   */
+  network: Network | Connection;
+}
+
+/**
+ * Transfer SOL or SPL tokens between accounts
+ *
+ * @param apiClient - The API client to use
+ * @param options - The transfer options
+ *
+ * @returns The transfer result
+ */
+export async function transfer(
+  apiClient: CdpOpenApiClientType,
+  options: TransferOptions,
+): Promise<SignatureResult> {
+  const connection = getOrCreateConnection({ networkOrConnection: options.network });
+
+  const tx =
+    options.token === "sol"
+      ? await getNativeTransfer({
+          connection,
+          from: options.from,
+          to: options.to,
+          amount: options.amount,
+        })
+      : await getSplTransfer({
+          connection,
+          from: options.from,
+          to: options.to,
+          mintAddress:
+            options.token === "usdc"
+              ? getUsdcMintAddress(await getConnectedNetwork(connection))
+              : options.token,
+          amount: options.amount,
+        });
+
+  const serializedTx = Buffer.from(tx.serialize()).toString("base64");
+
+  const signedTxResponse = await apiClient.signSolanaTransaction(options.from, {
+    transaction: serializedTx,
+  });
+
+  const decodedSignedTx = Buffer.from(signedTxResponse.signedTransaction, "base64");
+
+  const signature = await connection.sendRawTransaction(decodedSignedTx, {
+    skipPreflight: false,
+    maxRetries: 3,
+  });
+
+  return { signature };
+}
+
+type GetNativeTransferOptions = Omit<TransferOptions, "token" | "network"> & {
+  /**
+   * The connection to the Solana network
+   */
+  connection: Connection;
+};
+
+/**
+ * Get the instructions for a native SOL transfer
+ *
+ * @param options - The options for the native SOL transfer
+ *
+ * @param options.connection - The Solana connection
+ * @param options.from - The source address
+ * @param options.to - The destination address
+ * @param options.amount - The amount to transfer
+ *
+ * @returns The instructions for the native SOL transfer
```

Comment:
this still says instructions while we return transaction. Can we update it to say transaction?